### PR TITLE
Interface: Make annotation check optional

### PR DESCRIPTION
**Pull request by tiran**
Fixes: https://github.com/pyca/cryptography/issues/5774
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: src/cryptography/utils.py [hunk 1]
```diff
@@ -69,7 +69,21 @@ class InterfaceNotImplemented(Exception):
...
+        [
+            inspect.Parameter(
+                param.name,
+                param.kind,
+                default=param.default,
+                annotation=inspect.Parameter.empty,
+            )
+            for param in signature.parameters.values()
+        ]
+    )
```
Comment (stanislavlevin) [inline src/cryptography/utils.py:83]: What about `replace` methods?
for example,
```
return signature.replace(                                                                    
    parameters=[                                                                   
        param.replace(annotation=inspect.Parameter.empty)                          
        for param in signature.parameters.values()                                       
    ],                                                                             
    return_annotation=inspect.Signature.empty,                                     
) 
```

**Comment by alex** [comment]
Interesting, I hadn't realized anyone was using `register_interface` outside of our tree. WDYT @reaperhulk ?

**Comment by reaperhulk** [comment]
If we're breaking people on this we should definitely merge a fix. I'm fine with this approach.

**Comment by alex** [comment]
Note that this does change the default on whether it verifies sigs, I suppose that's fine?

**Review by reaperhulk — APPROVED** [review]

**Comment by reaperhulk** [comment]
Yeah I'm fine with that to be clear :)

**Pull request merged**